### ISSUE TITLE
Indentation in decorators not working properly

### ISSUE THREAD

**Issue by lanthaler** [eslint/typescript-eslint-parser#438]
**What version of TypeScript are you using?**
2.6.2

**What version of `typescript-eslint-parser` are you using?**
12.0.0

**What code were you trying to parse?**

```ts
import { Component, Vue } from 'vue-property-decorator'
import TForm from '@components/t-form.vue'

@Component({
  components: {
    't-form': TForm
  }
})
export default class extends Vue {
```

**What did you expect to happen?**
No linting errors

**What happened?**
Got the following errors

```ts
import { Component, Vue } from 'vue-property-decorator'
import TForm from '@components/t-form.vue'

@Component({
  components: {
    't-form': TForm    <--- *** Expected indentation of 2 spaces but found 4  indent ***
  }
})    <--- *** Expected indentation of 2 spaces but found 0  indent ***
export default class extends Vue {
```

**Comment by hartmut-co-uk** [comment]
+1

**Comment by bkotos** [comment]
+1

**Comment by phyllisstein** [comment]
This persists through `v14.0.0` with TypeScript `v2.8.0-rc`. Here is another sample of incorrectly flagged code:

```ts
@connect(
  (state: State) => ({
    appBarVisible: state.ui.appBarVisible,
  }),
)
class App extends React.Component<Props> {
```

This should not throw an error; instead, ESLint reports:

```
warning: Expected indentation of 4 spaces but found 2 (indent) at src/app/routes/app/app.tsx:19:1:
  17 |   (state: State) => ({
  18 |     appBarVisible: state.ui.appBarVisible,
> 19 |   }),
     | ^
  20 | )
  21 | class App extends React.Component<Props> {
  22 |   public shouldComponentUpdate(nextProps: Props): boolean {
```

This may or may not be related to broader issues I've noticed with the `indent` rule (q.v. #452).

**Comment by Nandiin** [comment]
1. Any update on this?
2. Any workaround for this?

**Comment by mcMickJuice** [comment]
@JamesHenry do you mean to say that a rule like `indent` shouldn't be the concern of a linter?

**Comment by zuzusik** [comment]
anyone was able to solve this somehow?

**Comment by archSeer** [comment]
I think it should be fine to disable this rule; eslint should handle
linting code, not formatting.

I disabled the rule and use prettier or some similar alternative after
eslint to handle indentation.

**Comment by zuzusik** [comment]
>eslint should handle linting code, not formatting.

@archSeer this is a bit of misconception 

https://eslint.org/docs/about/ says

> Code linting is a type of static analysis that is frequently used to find problematic patterns or code that doesn’t adhere to certain style guidelines.

**Comment by archSeer** [comment]
@zuzusik to each their own, I'd rather use eslint to catch problematic patterns than enforce indentation if git pre-commit hooks or https://editorconfig.org/ can do the same
